components/screens/contracts/ContractScreen.tsx

```diff
@@ -275,6 +305,44 @@ class ContractScreen extends React.Component<Props, State> {
     );
   }
 
+  /**
+   * Returns true if all fields of area detail values are filled
+   * @param areaDetailValues area detail values
+   */
+  private allFieldsFilled = (areaDetailValues: AreaDetail[]): boolean => {
+    for (const areaDetail of areaDetailValues) {
+      const { name, size, species, profitEstimation } = areaDetail;
+      if (!name || !size || !species || !profitEstimation) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Returns if contract proposed quantity is at least the total amount calculated from area details
+   * @param totalAmount total amount calculated from area details
+   */
```

Comment:
```suggestion
  /**
   * Returns if contract proposed quantity is at least the total amount calculated from area details
   * @param totalAmount total amount calculated from area details
   * @returns true if proposed quantity is at least the total amount, otherwise false
   */
```